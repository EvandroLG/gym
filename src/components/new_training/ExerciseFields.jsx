import React, { Component } from 'react';

export default class ExerciseFields extends Component {

  constructor(props) {
    super(props);

    this.state = {
      name: '',
      repetition: '',
      weight: '',
      set: '',
      youtube: ''
    };
  }

  componentDidMount() {
    if (this.nameInput) {
      this.nameInput.focus();
    }
  }

  onInputChange(e) {
    const key = e.target.id.split('_')[1];

    this.setState({
      [key]: e.target.value
    }, () => {
      this.props.onInputChange(this.props.index + '', this.state)
    });
  }

  render() {
    return (
      <div className="row form-group mt-1 d-flex">
        <div className="col">
          <label htmlFor={`exercise_name_${this.props.index}`}>Name</label>

          <input type="text" className="form-control"
            ref={ (input) => this.nameInput = input }
            id={`exercise_name_${this.props.index}`}
            onChange={this.onInputChange.bind(this)} />
        </div>

        <div className="col">
          <label htmlFor={`exercise_set_${this.props.index}`}>Set</label>

          <input type="text" className="form-control"
            id={`exercise_set_${this.props.index}`}
            onChange={this.onInputChange.bind(this)} />
        </div>

        <div className="col">
          <label htmlFor={`exercise_repetition_${this.props.index}`}>Repetition</label>

          <input type="text" className="form-control"
            id={`exercise_repetition_${this.props.index}`}
            onChange={this.onInputChange.bind(this)} />
        </div>

        <div className="col">
          <label htmlFor={`exercise_weight_${this.props.index}`}>Weight</label>

          <input type="text" className="form-control"
            id={`exercise_weight_${this.props.index}`}
            onChange={this.onInputChange.bind(this)} />
        </div>

        <div className="col">
          <label htmlFor={`exercise_youtube_${this.props.index}`}>Youtube</label>

          <input type="text" className="form-control"
            id={`exercise_youtube_${this.props.index}`}
            onChange={this.onInputChange.bind(this)} />
        </div>

        <div className="col align-self-center">
          <button type="button" className="btn btn-danger btn-sm"
            onClick={ () => this.props.onButtonRemoveExercise(this.props.index) }>
            Remove
          </button>
        </div>
      </div>
    )
  }
}
