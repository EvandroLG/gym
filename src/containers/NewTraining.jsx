import { connect } from 'react-redux';
import { setTitleTraining } from '../action';
import NewTraining from '../components/new_training/Index';

const mapStateToProps = ({ newTraining }) => {
  return {
    title: newTraining.title
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    onTitleUpdate: (value) => {
      dispatch(setTitleTraining(value));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(NewTraining);
