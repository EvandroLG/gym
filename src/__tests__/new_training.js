import React from 'react';
import { shallow } from 'enzyme';
import NewTraining from '../components/new_training/index';
import ExerciseFields from '../components/new_training/exercise_fields';

describe('new training', () => {
  const newTraining = shallow(<NewTraining />);
  const exerciseFields = shallow(<ExerciseFields />);

  it('new training component should render as expected', () => {
    verifySnapshot(newTraining);
  });

  it('exercise fields component should render as expected', () => {
    verifySnapshot(exerciseFields);
  });

  describe('events', () => {
    it('should create a box with correct form fields after click on add button', () => {
      expect(newTraining.find(ExerciseFields)).toHaveLength(0);
      newTraining.find('#add').simulate('click');
      expect(newTraining.find(ExerciseFields)).toHaveLength(1);
      newTraining.find('#add').simulate('click');
      expect(newTraining.find(ExerciseFields)).toHaveLength(2);
    });

    it('should delete a box when delete button was clicked', () => {
      //newTraining.find('.btn-danger').first().simulate('click', { preventDefault() {} });
      //expect(newTraining.find(ExerciseFields)).toHaveLength(1);
    });
  });
});
