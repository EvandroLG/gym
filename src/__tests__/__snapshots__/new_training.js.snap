// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new training exercise fields component should render as expected 1`] = `
<div
  className="row form-group mt-1 d-flex"
>
  <div
    className="col"
  >
    <label
      htmlFor="exercise_name_undefined"
    >
      Name
    </label>
    <input
      className="form-control"
      id="exercise_name_undefined"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="col"
  >
    <label
      htmlFor="exercise_set_undefined"
    >
      Set
    </label>
    <input
      className="form-control"
      id="exercise_set_undefined"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="col"
  >
    <label
      htmlFor="exercise_repetition_undefined"
    >
      Repetition
    </label>
    <input
      className="form-control"
      id="exercise_repetition_undefined"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="col"
  >
    <label
      htmlFor="exercise_weight_undefined"
    >
      Weight
    </label>
    <input
      className="form-control"
      id="exercise_weight_undefined"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="col"
  >
    <label
      htmlFor="exercise_youtube_undefined"
    >
      Youtube
    </label>
    <input
      className="form-control"
      id="exercise_youtube_undefined"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="col align-self-center"
  >
    <button
      className="btn btn-danger btn-sm"
      onClick={[Function]}
      type="button"
    >
      Remove
    </button>
  </div>
</div>
`;

exports[`new training new training component should render as expected 1`] = `
<NewTraining>
  <div>
    <h1>
      Register a new training
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="title"
          value=""
        >
          Title
        </label>
        <input
          className="form-control"
          id="title"
          onChange={[Function]}
          value=""
        />
        <button
          className="btn btn-primary btn-sm mt-1"
          id="add"
          onClick={[Function]}
          type="button"
        >
          Add Exercise
        </button>
      </div>
      <input
        className="btn btn-success btn-block mt-2"
        type="submit"
        value="Register it!"
      />
    </form>
  </div>
</NewTraining>
`;
